workspace/__lib__/xod/bits/shift-left/patch.cpp

```diff
@@ -7,7 +7,15 @@ struct State {
 {{ GENERATED_CODE }}
 
 void evaluate(Context ctx) {
-    uint8_t x = (uint8_t)getValue<input_BYTE>(ctx);
-    uint8_t n = (uint8_t)getValue<input_N>(ctx);
-    emitValue<output_OUT>(ctx, x << n);
+    int32_t x = (int32_t)getValue<input_IN>(ctx);
+    int32_t n = (int32_t)getValue<input_N>(ctx);
+    int32_t b = 0;
+    if (n < 0) {
+      b = 0;
+    } else if (n > 31) {
+      b = 31;
+    } else {
+      b = n;
+    }
```

Comment:
https://www.arduino.cc/reference/en/language/functions/math/constrain/ ?

Reply:
I did not use `constrain` to avoid bugs on some other platforms that did not define it.